app/modules/crm_questionnaire/server.R

```diff
@@ -0,0 +1,85 @@
+library(shiny)
+
+parse_params <- function(params_str) {
+  params <- strsplit(params_str, ";")[[1]]
+  param_list <- lapply(params, function(p) strsplit(p, "=")[[1]])
+  names(param_list) <- sapply(param_list, `[`, 1)
+  sapply(param_list, `[`, 2)
+}
+
+questions <- read.csv("app/data/questionnaire_inputs/crm_q_database.csv")
+
+server <- function(input, output, session) {
+
+  # Load button
+  observe({
+    in_file <- input$file_upload
+
+    if (is.null(in_file)) {
+      return(NULL)
+    }
+
+    ext <- tools::file_ext(in_file$datapath)
+
+    if (ext == "csv") {
+      user_responses <- read.csv(in_file$datapath)
+    } else if (ext == "rds") {
+      user_responses <- readRDS(in_file$datapath)
+    }
+    for (i in seq_len(nrow(user_responses))){
+      updateNumericInput(session,
+                         inputId = user_responses$inputId[i],
+                         value = user_responses$value[i])
+    }
+  })
+
+  output$questionsUI <- renderUI({
+    tagList(
+      lapply(seq_len(nrow(questions)), function(i) {
+        question <- questions[questions$q_number == i, ]
+        params <- parse_params(question$params)
+
+        switch(question$q_type,
+          radioButtons = radioButtons(inputId = question$q_variable,
+                                      label = question$q_text,
+                                      choices = strsplit(params[["choices"]], ",")[[1]],
+                                      width = 500),
+          numeric = numericInput(inputId = question$q_variable,
+                                 label = question$q_text,
+                                 min = as.numeric(params[["min"]]),
+                                 value = 0,
```

Comment:
I think this needs to be edited to check if answer_type is numeric or numeric_bounded so that we can specify max when given in the question database for 'nuneric_bounded', but so we aren't specifying anythig for answers that are just 'numeric'. I'll try to do this this afternoon. Same in tpt server.

Reply:
I think that we can solve this by having params for any 'numeric' answer type read as 'min=x,max=NA,step=y' and then include line between 49 and 50 as max = as.numeric(params[["max"]]).